src/js/components/Form/__tests__/Form-test-uncontrolled.js

```diff
@@ -302,75 +392,212 @@ describe('Form uncontrolled', () => {
     });
+    getByText('submit').focus();
     fireEvent.click(getByText('submit'));
     expect(queryAllByText('required')).toHaveLength(1);
 
-    // name field has new error and email field still
-    // has required error message
+    // name field has new error and email field still has required error message
+    getByPlaceholderText('name').focus();
     fireEvent.change(getByPlaceholderText('name'), {
       target: { value: 'a' },
     });
+    getByText('submit').focus();
     fireEvent.click(getByText('submit'));
     expect(queryByText('required')).toBeTruthy();
     expect(queryByText('must be >1 character')).toBeTruthy();
 
     //  new value in name does not remove the error message in email
+    getByPlaceholderText('name').focus();
     fireEvent.change(getByPlaceholderText('name'), {
       target: { value: 'abc' },
     });
+    expect(onFocus).toBeCalledTimes(6);
     expect(queryByText('required')).toBeTruthy();
     expect(queryByText('must be >1 character')).toBe(null);
   });
 
+  test('uncontrolled without name', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, container } = render(
+      <Form onSubmit={onSubmit}>
+        <FormField>
+          <TextInput a11yTitle="test" placeholder="test input" />
+        </FormField>
+        <Button type="submit" primary label="Submit" />
+      </Form>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'v' },
+    });
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('Submit'));
+    expect(onSubmit).toBeCalledTimes(1);
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('uncontrolled reset without value', () => {
+    const onChange = jest.fn();
+    const { getByPlaceholderText, getByText, queryByText } = render(
+      <Grommet>
+        <Form onChange={onChange}>
+          <FormField
+            name="test"
+            required
+            placeholder="test input"
+            a11yTitle="test"
+          />
+          <Button type="reset" primary label="Reset" />
+        </Form>
+      </Grommet>,
+    );
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'Input has changed' },
+    });
+    expect(onChange).toBeCalledTimes(1);
+    fireEvent.click(getByText('Reset'));
+    expect(queryByText('Input has changed')).toBeNull();
+  });
+
+  test('disabled', () => {
+    const onSubmit = jest.fn();
+    const { getByPlaceholderText, getByText, container } = render(
+      <Grommet>
+        <Form onSubmit={onSubmit}>
+          <FormField disabled>
+            <TextInput a11yTitle="test" placeholder="test input" />
+          </FormField>
+          <Button type="submit" primary label="Submit" />
+        </Form>
+      </Grommet>,
+    );
+    fireEvent.change(getByPlaceholderText('test input'), {
+      target: { value: 'v' },
+    });
```

Comment:
Similar comment regarding assertion vs snapshot.

Reply:
Removed snapshot and added assertion